src/Middleware/Diagnostics/src/DiagnosticsLoggerExtensions.cs

```diff
@@ -20,7 +21,9 @@ internal static class DiagnosticsLoggerExtensions
             LoggerMessage.Define(LogLevel.Error, new EventId(3, "Exception"), "An exception was thrown attempting to execute the error handler.");
 
         private static readonly Action<ILogger, Exception?> _errorHandlerNotFound =
-            LoggerMessage.Define(LogLevel.Warning, new EventId(4, "HandlerNotFound"), "No exception handler was found, rethrowing original exception.");
+            LoggerMessage.Define(LogLevel.Warning, new EventId(4, "HandlerNotFound"), $"The exception handler configured on {nameof(ExceptionHandlerOptions)} produced a 404 status response. " +
```

Comment:
You shouldn't be using interpolation in logger messages.

Reply:
Even for nameof? These aren't variables.

Reply:
Not with the {} syntax

Reply:
Fair enough.

Reply:
Need an analyzer for this. We have this elsewhere in our repo: https://github.com/dotnet/aspnetcore/blob/main/src/Mvc/Mvc.Core/src/MvcCoreLoggerExtensions.cs#L294.

Reply:
There's on we're shipping in 6.0